openslides/users/migrations/0004_groups.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.9.7 on 2016-08-01 14:54
+from __future__ import unicode_literals
+
+from django.contrib.auth.models import Group
+from django.db import migrations, models
+
+def migrate_groups_and_user_permissions(apps, schema_editor):
+    """
+    This function migrates the database to the new groupslogic:
+    - Rename Group 'Anonymus' (pk=1) to 'Default'
+    - Rename Group 'Registered users' (pk=2) to 'oldRegisteres'
+    - Add all users who are not in any group to this group (pk=2)
+    - Add all permissions of 'old'Registeres' to all other groups (exclude 'Default')
+
+    But ony do this migration if:
+    - there are users in the database
+    - the name of the first group is 'Guests'.
+    """
+    User = apps.get_model('users', 'User')
+
+    if Group.objects.all().count() != 0:
+        if Group.objects.get(pk=1).name == 'Guests':
```

Comment:
Line 23 should test `exists()`, line should not use and uncatched `get` because it is possible that the groups are deleted in the database if someone manipulated it. Our migration file should not break in this case. Last point: My question was about the check in users/signals.py:14 not about this line.